beacon_node/execution_layer/src/lib.rs

```diff
@@ -795,51 +837,73 @@ impl<T: EthSpec> ExecutionLayer<T> {
     ///
     /// The result will be returned from the first node that returns successfully. No more nodes
     /// will be contacted.
-    pub async fn get_payload<Payload: AbstractExecPayload<T>>(
+    #[allow(clippy::too_many_arguments)]
+    pub async fn get_payload(
         &self,
         parent_hash: ExecutionBlockHash,
         payload_attributes: &PayloadAttributes,
         forkchoice_update_params: ForkchoiceUpdateParameters,
         builder_params: BuilderParams,
         current_fork: ForkName,
         spec: &ChainSpec,
-    ) -> Result<BlockProposalContents<T, Payload>, Error> {
-        let payload_result = match Payload::block_type() {
-            BlockType::Blinded => {
+        block_production_version: BlockProductionVersion,
+    ) -> Result<BlockProposalContentsType<T>, Error> {
+        let payload_result_type = match block_production_version {
+            BlockProductionVersion::V3 => match self
+                .determine_and_fetch_payload(
+                    parent_hash,
+                    payload_attributes,
+                    forkchoice_update_params,
+                    builder_params,
+                    current_fork,
+                    spec,
+                )
+                .await
+            {
+                Ok(payload) => payload,
+                Err(e) => {
+                    metrics::inc_counter_vec(
+                        &metrics::EXECUTION_LAYER_GET_PAYLOAD_OUTCOME,
+                        &[metrics::FAILURE],
+                    );
+                    return Err(e);
+                }
+            },
+            BlockProductionVersion::BlindedV2 => {
                 let _timer = metrics::start_timer_vec(
                     &metrics::EXECUTION_LAYER_REQUEST_TIMES,
                     &[metrics::GET_BLINDED_PAYLOAD],
                 );
-                self.get_blinded_payload(
+                self.determine_and_fetch_payload(
                     parent_hash,
                     payload_attributes,
                     forkchoice_update_params,
                     builder_params,
                     current_fork,
                     spec,
                 )
-                .await
+                .await?
             }
-            BlockType::Full => {
-                let _timer = metrics::start_timer_vec(
-                    &metrics::EXECUTION_LAYER_REQUEST_TIMES,
-                    &[metrics::GET_PAYLOAD],
-                );
```

Comment:
I think maybe we should keep this metric in the FullV2 case. Or maybe we should move it lower in the stack to where we _actually_ call `getPayload`?